pinot-common/src/main/java/org/apache/pinot/pql/parsers/PinotQuery2BrokerRequestConverter.java

```diff
@@ -0,0 +1,362 @@
+import org.apache.pinot.common.request.Expression;
+import org.apache.pinot.common.request.ExpressionType;
+import org.apache.pinot.common.request.FilterOperator;
+import org.apache.pinot.common.request.FilterQuery;
+import org.apache.pinot.common.request.FilterQueryMap;
+import org.apache.pinot.common.request.Function;
+import org.apache.pinot.common.request.GroupBy;
+import org.apache.pinot.common.request.Literal;
+import org.apache.pinot.common.request.PinotQuery;
+import org.apache.pinot.common.request.QuerySource;
+import org.apache.pinot.common.request.QueryType;
+import org.apache.pinot.common.request.Selection;
+import org.apache.pinot.common.request.SelectionSort;
+import org.apache.pinot.pql.parsers.pql2.ast.FilterKind;
+
+
+public class PinotQuery2BrokerRequestConverter {
+
+  static Map<FilterKind, FilterOperator> filterOperatorMapping;
+
+  public BrokerRequest convert(PinotQuery pinotQuery) {
+    BrokerRequest brokerRequest = new BrokerRequest();
+
+    //Query Source
+    QuerySource querySource = new QuerySource();
+    querySource.setTableName(pinotQuery.getDataSource().getTableName());
+    brokerRequest.setQuerySource(querySource);
+
+    handleFilter(pinotQuery, brokerRequest);
+
+    //Handle select list
+    handleSelectList(pinotQuery, brokerRequest);
+
+    //Handle order by
+    handleOrderBy(pinotQuery, brokerRequest);
+
+    //Handle group by
+    handleGroupBy(pinotQuery, brokerRequest);
+
+    //Query Type
+    QueryType queryType = new QueryType();
+    if (brokerRequest.getAggregationsInfo() != null && brokerRequest.getAggregationsInfo().size() > 0) {
+      if (brokerRequest.getGroupBy() != null) {
+        queryType.setHasGroup_by(true);
+      } else {
+        queryType.setHasAggregation(true);
+      }
+    } else {
+      queryType.setHasSelection(true);
+    }
+    // Commenting this out since the current code does not set it.
+    // brokerRequest.setQueryType(queryType);
+
+    //TODO: these should not be part of the query?
+    //brokerRequest.setEnableTrace();
+    //brokerRequest.setDebugOptions();
+    brokerRequest.setQueryOptions(pinotQuery.getQueryOptions());
+    //brokerRequest.setBucketHashKey();
+    //brokerRequest.setDuration();
+
+    return brokerRequest;
+  }
+
+  private void handleOrderBy(PinotQuery pinotQuery, BrokerRequest brokerRequest) {
+    if (brokerRequest.getSelections() == null || pinotQuery.getOrderByList() == null) {
+      return;
+    }
+    List<SelectionSort> sortSequenceList = new ArrayList<>();
+    final List<Expression> orderByList = pinotQuery.getOrderByList();
+    for (Expression orderByExpr : orderByList) {
+      SelectionSort selectionSort = new SelectionSort();
+      if (orderByExpr.getFunctionCall().getOperator().equalsIgnoreCase("ASC")) {
+        selectionSort.setIsAsc(true);
+      } else {
+        selectionSort.setIsAsc(false);
+      }
+      selectionSort.setColumn(orderByExpr.getFunctionCall().getOperands().get(0).getIdentifier().getName());
```

Comment:
Is size of operands guaranteed to be > 0? If not, better to handle gracefully.

Reply:
Actually, that line is buggy. It assumes that orderby is always a function. Will fix that. Good catch.